### PR TITLE
NOW-PROTO corrections

### PR DESCRIPTION
**Pull request by pacmancoder**
Proposed changes/questions for the draft NOW-PROTO specification

### CODE REVIEW

File: docs/NOW-spec.md [hunk 1]
```diff
@@ -519,32 +519,6 @@ The NOW_STRING256 structure is used to represent variable-length strings of up t
 
 #### NOW_BUFFER
 
-The NOW_BUFFER structure is used to represent variable-length buffers.
```
Comment (pacmancoder) [inline docs/NOW-spec.md:522]: From my pont of view, there is not much value in this type when we already have more compact `NOW_VARBUF`

File: docs/NOW-spec.md [hunk 2]
```diff
@@ -639,10 +613,7 @@ The NOW_HEADER structure is the header common to all NOW protocol messages.
 **msgFlags (2 bytes)**: The message flags, specific to the message type and class.
 
 #### NOW_STATUS
-
-##### NOW_VARU16
-
-The NOW_VARU16 structure is used to encode unsigned integer values in the range [0, 0x7FFF].
```
Comment (pacmancoder) [inline docs/NOW-spec.md:645]: must likely this section was added here by accident

File: docs/NOW-spec.md [hunk 3]
```diff
@@ -729,7 +700,10 @@ The NOW_VARU16 structure is used to encode unsigned integer values in the range
 | NOW_SYSTEM_INFO_RSP_ID<br>0x02 | NOW_SYSTEM_INFO_RSP_MSG |
 | NOW_SYSTEM_SHUTDOWN_ID<br>0x03 | NOW_SYSTEM_SHUTDOWN_MSG |
 
+<!-- TODO: Define NOW_SYSTEM_INFO_REQ_MSG, NOW_SYSTEM_INFO_RSP_MSG   -->
```
Comment (pacmancoder) [inline docs/NOW-spec.md:703]: Structures are not defined yet -- should we remove message kinds for them or they are to be defined?

File: docs/NOW-spec.md [hunk 4]
```diff
@@ -1084,20 +1058,20 @@ The NOW_EXEC_CAPSET_MSG message is sent to advertise capabilities.
 
 **msgType (1 byte)**: The message type (NOW_EXEC_CAPSET_MSG_ID).
 
-**msgFlags (2 bytes)**: The message flags.
-
-**execStyles (4 bytes)**: A 32-bit unsigned integer containing the supported execution styles.
+**msgFlags (2 bytes)**: A 16-bit unsigned integer containing the supported execution styles.
```
Comment (pacmancoder) [inline docs/NOW-spec.md:1061]: I think we could use flags fields for styles

File: docs/NOW-spec.md [hunk 5]
```diff
@@ -1291,9 +1265,6 @@ The NOW_EXEC_DATA_MSG message is used to send input/output data as part of a rem
         <tr>
             <td colspan="32">sessionId</td>
         </tr>        
-        <tr>
```
Comment (pacmancoder) [inline docs/NOW-spec.md:1294]: We could use `NOW_VARBUF` here

File: docs/NOW-spec.md [hunk 6]
```diff
@@ -1318,9 +1289,7 @@ The NOW_EXEC_DATA_MSG message is used to send input/output data as part of a rem
 
 **sessionId (4 bytes)**: A 32-bit unsigned integer containing a unique remote execution session id.
 
-**size (4 bytes)**: A 32-bit unsigned integer containing input/output data size.
-
-**data (variable)**: The input/output data with the size specified by the size field.
+**data (variable)**: The input/output data represented as `NOW_VARBUF`
```
Comment (pacmancoder) [inline docs/NOW-spec.md:1292]: Should we use `NOW_VARBUF` or `NOW_LRGBUF` here?

File: docs/NOW-spec.md [hunk 7]
```diff
@@ -1365,6 +1334,9 @@ The NOW_EXEC_RUN_MSG message is used to send a run request. This request type ma
 
 **command (variable)**: A NOW_VARSTR structure containing the command to execute.
 
+#### NOW_EXEC_CMD_MSG
+<!-- TODO: Define CMD message -->
```
Comment (pacmancoder) [inline docs/NOW-spec.md:1338]: Not defined -- should be keep it? We already have `NOW_EXEC_BATCH_MSG`, what is the difference?

File: docs/NOW-spec.md [hunk 8]
```diff
@@ -1412,11 +1384,11 @@ The NOW_EXEC_PROCESS_MSG message is used to send a Windows [CreateProcess()](htt
 
 **sessionId (4 bytes)**: A 32-bit unsigned integer containing a unique remote execution session id.
 
-**filename (variable)**: A NOW_STRING structure containing the file name. Corresponds to the lpApplicationName parameter.
+**filename (variable)**: A NOW_VARSTR structure containing the file name. Corresponds to the lpApplicationName parameter.
```
Comment (pacmancoder) [inline docs/NOW-spec.md:1387]: There is no `NOW_STRING` in this specification (section empty). I think `NOW_VARSTR` suits perfect here

File: docs/NOW-spec.md [hunk 9]
```diff
@@ -1537,6 +1509,12 @@ The NOW_EXEC_WINPS_MSG message is used to execute a remote Windows PowerShell (p
         <tr>
             <td colspan="32">command (variable)</td>
         </tr>
+        <tr>
```
Comment (pacmancoder) [inline docs/NOW-spec.md:1512]: missing fields from graphical representation

**Comment by pacmancoder** [comment]
Additional points:
- Do we need to keep `NOW_VARU16, NOW_VARU64, NOW_VARI16, NOW_VARI32, NOW_VARI64` types? They are not used anywhere in the messages (yet)
- Do we need `NOW_LGRSTR`/`NOW_LRGBUF` types? Currently they are not used in the messages

File: docs/NOW-spec.md (hunk 3 above)
Comment (CBenoit) [reply docs/NOW-spec.md:703]: Maybe a leftover from the original document :thinking:

**Review by CBenoit — APPROVED** [review]
LGTM!

**Pull request merged**